be/src/storage/rowset/beta_rowset_writer.cpp

```diff
@@ -291,13 +297,8 @@ Status HorizontalBetaRowsetWriter::flush_chunk(const vectorized::Chunk& chunk) {
         break;
     case FlushChunkState::UPSERT:
         break;
-    default: {
-        std::string msg =
-                "multi-segment only supported by pure upsert/delete, so pure upsert after another "
-                "operation is illegal";
-        LOG(WARNING) << msg;
-        return Status::Cancelled(msg);
-    }
+    default:
+        return _mixed_segment_delfile_not_supported();
```

Comment:
_mixed_segment_delfile_not_supported()
```
Status HorizontalBetaRowsetWriter::_mixed_segment_delfile_not_supported() {
    std::string msg = strings::Substitute(
            "multi-segment rowset do not support mixing upsert and delete tablet:$0 txn:$1 #seg:$2 #delfile:$3 "
            "#upsert:$4 #del:$5",
            _context.tablet_id, _context.txn_id, _num_segment, _num_delfile, _num_rows_written, _num_rows_del);
    LOG(WARNING) << msg;
    return Status::Cancelled(msg);
}
```
this function do two things, one is dump the information of why here is not support, two is return a cancel status.
and from this line:
```
    return _mixed_segment_delfile_not_supported();
```
code reader can not know how is going on, why it so-called "not support", reader need look insight into the _mixed_segment_delfile_not_supported() implement.

how about this, when a function print some infos about xxx, we call it "dump_xxx_info" or "dump_info_xxx" (there some log print function like this, they can be normalized as well)
for example, here can be:
```
    return Status::Cancelled(_dump_mixed_segment_delfile_not_supported_info());
```
like rocksdb
https://github.com/facebook/rocksdb/blob/49628c9a83d337ea7b01f7ad3bdffa340e013913/db/compaction/compaction_picker_universal.cc#L654-L657
```
        char file_num_buf[256];
        skipping_sr->DumpSizeInfo(file_num_buf, sizeof(file_num_buf), loop);
        ROCKS_LOG_BUFFER(log_buffer_, "[%s] Universal: Skipping %s",
                         cf_name_.c_str(), file_num_buf);
```